### ISSUE TITLE
Misalignment of chart profile on Mac and iOS devices

### ISSUE THREAD

**Issue by riccardoklinger** [Raruto/leaflet-elevation#123]
unfortunately the mobile view shows some misalignement for the profile:
on iOS with Firefox/Safari
![grafik](https://user-images.githubusercontent.com/7377897/128267445-63083f19-2baa-448c-836a-264018cf0efb.png)
on Web-preview for mobile view.
![grafik](https://user-images.githubusercontent.com/7377897/128267398-9926e919-c64c-4c7d-a36a-eceebed01ade.png)
source can be found here: 
https://riccardoklinger.github.io/POTB/#3
any ideas? is it a known bug?

**Comment by riccardoklinger** [comment]
same issue on desktop Safari:
![grafik](https://user-images.githubusercontent.com/7377897/128273016-ea50dc78-c175-4e54-9927-c96ef7756952.png)
will debvug there.

**Comment by riccardoklinger** [comment]
visualisation under 0.3.9 was working fine in iOS based browser (Safari and firefox iOS)

**Comment by Raruto** [comment]
Hi Riccardo,
I have no problem with the following configuration (have you also tested on physical devices? are you on mac os or related devices?):

```log
iPhone X/XS iOS12 - 375x812 - portrait - DPR:3

Throttling: No Limit

UA: Mozilla/5.0 (iPhone; CPU iPhone OS 12_0 like Mac OS X) AppleWebKit/604.1.38 (KHTML, like Gecko) Version/12.0 Mobile/15A372 Safari/604.1

Touch simulation: enabled
```

![mobile-view](https://user-images.githubusercontent.com/9614886/128305146-b4a936d4-642b-4768-a99b-64305e6a76ab.png)

_Have a nice day,
Raruto_

**Comment by riccardoklinger** [comment]
Yes, I’ve tried it on my iPhone as well as macOS with Safari.

Von meinem iPhone gesendet

**Comment by Raruto** [comment]
Ok thanks for the info,

however please note that I am unable to verify it, [as usual](https://github.com/Raruto/leaflet-elevation/issues/108), I leave this task to anyone who is keenly interested in fixing this issue ..

_Have a nice day,
Raruto_

**Comment by skirridsystems** [comment]
No, no reports here, but I'm not using the latest version yet in my plugin. I'm due for an update soon so I'll check it on BrowserStack then.

**Comment by qwertydude** [comment]
Hi

Am having same problem on Mac and iPhone devices.

Misalignment occurs on Mac Safari, and Chrome, Firefox and Safari on iOS devices.

Also, the tooltip is appearing **behind** the elevation graph on Safari and Chrome but **not** Firefox on Mac. (See below)

And it is appearing behind it on Safari, Chrome and Firefox on iOS devices.

I've tested with older distributions, and **both** these issues first appear in Leaflet Elevation **v1.6.5**

![CleanShot 2021-08-24 at 18 11 16@2x](https://user-images.githubusercontent.com/129821/130581319-6b85c87c-2508-4af8-a398-4623dc92e4e6.jpg)

![CleanShot 2021-08-24 at 18 10 08@2x](https://user-images.githubusercontent.com/129821/130581200-e37425f0-9027-4e30-8187-783a9f350772.jpg)

![CleanShot 2021-08-24 at 18 12 05@2x](https://user-images.githubusercontent.com/129821/130581463-2b84734d-14ba-4f7a-bf0a-8dbcb0df62b6.jpg)

(Devices are MacBook Air M1,. iPhone X, iPad Air (3rd Gen). All running latest OS versions)

**Comment by riccardoklinger** [comment]
I'Ve solved this by using:
    // Render chart profiles as Canvas or SVG Paths
    preferCanvas: false,

**Comment by Raruto** [comment]
Thanks Riccardo for reporting.

So, to anyone with an apple device, do your tests with the following option enabled:

```js
// Render chart profiles as Canvas or SVG Paths
preferCanvas: true,
```
Related issues:

- https://github.com/Raruto/leaflet-elevation/issues/108
- https://github.com/Raruto/leaflet-elevation/issues/116

From what i see, this could still be the cause: https://stackoverflow.com/questions/63799631/svg-foreignobject-render-issue-in-chrome-and-safari

_Have a nice day,
Raruto_

**Comment by vutle** [comment]
I can confirm that the display text on chart correctly shown on firefox but it is behind the chart shaded area in chrome.

Firefox
![image](https://user-images.githubusercontent.com/6241928/133084383-1224ec3d-921f-41c7-9592-5daa026b3835.png)

Chrome + Microsoft Edge.
![image](https://user-images.githubusercontent.com/6241928/133084463-af7dc131-5569-4298-b935-fd072f08c459.png)
